sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/openai/properties/AzureOpenAIProperties.java

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.cloud.autoconfigure.implementation.openai.properties;
+
+import com.azure.ai.openai.OpenAIServiceVersion;
+import com.azure.spring.cloud.autoconfigure.implementation.properties.core.AbstractAzureHttpConfigurationProperties;
+import com.azure.spring.cloud.service.implementation.openai.OpenAIClientProperties;
+
+/**
+ * Configuration properties for Azure OpenAI.
+ */
+public class AzureOpenAIProperties extends AbstractAzureHttpConfigurationProperties implements OpenAIClientProperties {
+
+    public static final String PREFIX = "spring.cloud.azure.openai";
+
+    /**
+     * Endpoint of the Azure OpenAI. For instance, 'https://{azure-openai-name}.openai.azure.com/'.
+     */
+    private String endpoint;
+
+    /**
+     * Azure OpenAI service version used when making API requests.
+     */
+    private OpenAIServiceVersion serviceVersion;
+
+    /**
+     * The API key to authenticate the non-Azure OpenAI service (https://platform.openai.com/docs/api-reference/authentication).
+     */
+    private String nonAzureOpenAIKey;
```

Comment:
what does the configuration property for the `nonAzureOpenAIKey` look like in the application properties file?

Reply:
"spring.cloud.azure.openai.non-azure-openai-key=xxx"